geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala

```diff
@@ -11,124 +11,191 @@ package org.locationtech.geomesa.jobs.mapreduce
 import org.apache.accumulo.core.security.Authorizations
 import org.apache.accumulo.core.util.{Pair => AccPair}
-import org.apache.commons.collections.map.CaseInsensitiveMap
 import org.apache.hadoop.io.{Text, Writable}
 import org.apache.hadoop.mapreduce._
-import org.apache.hadoop.security.token.Token
-import org.geotools.data.{DataStoreFinder, Query}
-import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.data.Query
 import org.geotools.filter.text.ecql.ECQL
 import org.locationtech.geomesa.accumulo.AccumuloProperties.AccumuloMapperProperties
-import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloDataStoreParams}
-import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
-import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
-import org.locationtech.geomesa.index.api.GeoMesaFeatureIndex
-import org.locationtech.geomesa.index.conf.QueryHints.RichHints
+import org.locationtech.geomesa.accumulo.data.{AccumuloDataStore, AccumuloDataStoreParams, AccumuloQueryPlan}
+import org.locationtech.geomesa.index.api.QueryPlan.ResultsToFeatures
 import org.locationtech.geomesa.jobs.GeoMesaConfigurator
 import org.locationtech.geomesa.jobs.accumulo.AccumuloJobUtils
-import org.locationtech.geomesa.utils.index.IndexMode
-import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.locationtech.geomesa.jobs.mapreduce.GeoMesaAccumuloInputFormat.{GeoMesaRecordReader, GroupedSplit}
+import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
+import org.locationtech.geomesa.utils.io.WithStore
+import org.opengis.feature.simple.SimpleFeature
 import org.opengis.filter.Filter
 
-import scala.collection.JavaConversions._
 import scala.collection.mutable.ArrayBuffer
 
+/**
+  * Input format that allows processing of simple features from GeoMesa based on a CQL query
+  */
+class GeoMesaAccumuloInputFormat extends InputFormat[Text, SimpleFeature] with LazyLogging {
+
+  import scala.collection.JavaConverters._
+
+  private val delegate = new AccumuloInputFormat()
+
+  /**
+    * Gets splits for a job.
+    *
+    * Our delegated AccumuloInputFormat creates a split for each range - because we set a lot of ranges in
+    * geomesa, that creates too many mappers. Instead, we try to group the ranges by tservers. We use the
+    * location assignment of the tablets to tservers to determine the number of splits returned.
+    */
+  override def getSplits(context: JobContext): java.util.List[InputSplit] = {
+    val accumuloSplits = delegate.getSplits(context)
+    // Get the appropriate number of mapper splits using the following priority
+    // 1. Get splits from AccumuloMapperProperties.DESIRED_ABSOLUTE_SPLITS (geomesa.mapreduce.splits.max)
+    // 2. Get splits from #tserver locations * AccumuloMapperProperties.DESIRED_SPLITS_PER_TSERVER (geomesa.mapreduce.splits.tserver.max)
+    // 3. Get splits from AccumuloInputFormat.getSplits(context)
```

Comment:
Is this priority listed in the docs?

Reply:
I think so here: https://www.geomesa.org/documentation/user/accumulo/configuration.html#map-reduce-input-splits-properties
This didn't change, I just moved the code around a bit.